enclave-manager/web/src/emui/enclaves/enclave/service/logs/ServiceLogs.tsx

```diff
@@ -20,37 +20,55 @@ type ServiceLogsProps = {
   service: ServiceInfo;
 };
 
+// @ts-ignore
+export async function reTryCatch(callback, times: number = 1, isRetry: boolean = false) {
+  try {
+    return await callback(isRetry);
+  } catch (error) {
+    if (times > 0) {
+      console.info(`retrying another ${times} times`);
+      return await reTryCatch(callback, times - 1, true);
+    } else {
+      console.info("retry: giving up and throwing error");
+      throw error;
+    }
+  }
+}
+
 export const ServiceLogs = ({ enclave, service }: ServiceLogsProps) => {
   const kurtosisClient = useKurtosisClient();
   const [logLines, setLogLines] = useState<LogLineProps[]>([]);
 
   useEffect(() => {
-    let cancelled = false;
+    let canceled = false;
     const abortController = new AbortController();
-    (async () => {
+    if (kurtosisClient) {
```

Comment:
we are guaranteed to have a `kurtosisClient` - `useKurtosisClient` always provides us an instantiated, ready to go, client - so this check isn't needed.